cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/AggregatingFunctionsTest.scala

```diff
@@ -386,5 +394,92 @@ class AggregatingFunctionsTest extends DocumentingTest {
+          #Aggregation expressions are also allowed to be more complex, where the result of one or more aggregation functions
+          #are input arguments to other expressions.
+          #For instance, `0.1 * (SUM(x.a) / COUNT(x.b))` is an aggregation expression that contains two aggregation functions,
+          #`SUM( )` with `x.a` as its argument and `COUNT( )` with `x.b` as its argument.
+          #Both are input arguments to the division expression.
+          #""".stripMargin(
+          '#'))
+
+      p(
+        """For aggregation expressions to be correctly computable for the buckets formed by the grouping key(s), they have to fulfill some requirements.
+          #Specifically, each sub expression in an aggregation expression has to be either:
+          #
+          #* an aggregation function, e.g. `SUM(x.a)`,
+          #* a constant, e.g. `0.1`,
+          #* a parameter, e.g. `$param`,
+          #* a grouping key, e.g. the `a` in `RETURN a, count(*)`
+          #* a local variable, e.g. the `x` in  `count(*) + size([ x IN range(1, 10) | x ])`, or
+          #* a subexpression, all whose operands are operands allowed in an aggregation expression.
+          #""".stripMargin('#'))
+
+      section("Examples of aggregation expressions.", "grouping-key-examples") {
+        p("Simple aggregation without any grouping keys.")
+        query("""MATCH (p: Person) RETURN max(p.age)""",
+          ResultAssertions((r) => {
+            r.toList.head("max(p.age)") should equal(44L)
+          })) {
+          resultTable()
+        }
+        p("Addition of an aggregation and a constant, without any grouping keys.")
+        query("""MATCH (p: Person) RETURN max(p.age) + 1""",
+          ResultAssertions((r) => {
+            r.toList.head("max(p.age) + 1") should equal(45L)
+          })) {
+          resultTable()
+        }
+        p("Subtraction of a property access and an aggregation. Note that `n` is a grouping key.")
+        query("""MATCH (n: Person{name:"A"})-[:KNOWS]-(f:Person) RETURN n, n.age - max(f.age)""",
+          ResultAssertions((r) => {
+            r.toList.size shouldBe 1
+            r.head.get("n.age - max(f.age)") should equal(Some(-31L))
+          })) {
+          resultTable()
+        }
+        p("Subtraction of a property access and an aggregation. Note that `n.age` is a grouping key.")
+        query("""MATCH (n: Person{name:"A"})-[:KNOWS]-(f:Person) RETURN n.age, n.age - max(f.age)""",
+          ResultAssertions((r) => {
+            r.toList should equal(List(Map("n.age" -> 13L, "n.age - max(f.age)" -> -31)))
+          })) {
+          resultTable()
+        }
+
+        p(
+          """
+            #Grouping keys themselves can be complex expressions.
+            #For better query readability, Cypher only recognizes sub-expressions in aggregation expressions to be grouping key if the grouping key is either:
```

Comment:
Typo: The grammatical number of "to be grouping key" is not right.
```suggestion
            #For better query readability, Cypher only recognizes sub-expressions in aggregation expressions to be grouping keys if the grouping key is either:
```

Alternatively, one can fully switch to singular: 
"Cypher only recognizes a sub-expression in aggregation expressions as a grouping key if the grouping key is either"

To the side, I seem to like "recognizes ... as" better than "recognizes ... to be". But there are probably people more word-smithy then me to decide about such editorial details.